sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructField.scala

```diff
@@ -141,6 +141,10 @@ case class StructField(
     }
   }
 
+  private def getDDLDefault = getCurrentDefaultValue()
+    .map(" DEFAULT " + _)
```

Comment:
This looks a little weak. In case of STRING, we don't add quotation? If not, it looks like `DEFAULT THIS IS A DEFAULT VALUE`.

Reply:
Good question, I was originally confused about this as well :) the default value is exactly the string that appeared in the CREATE TABLE or ALTER TABLE statement. So, for example, if we had `CREATE TABLE t (a STRING DEFAULT 'str')`, then this column metadata has the value `'str'` (including the single-quotes). So we have to print them back exactly as-is here (e.g. I tried adding `escapeSingleQuotedString` like `getDDLComment` below, but it added extra escaping which we did not want).

The extra test cases you suggested should show that this works even for different types of string-typed defaults.